### PR TITLE
[6.3.x] test fixes

### PR DESCRIPTION
**Pull request by psiroky**

### CODE REVIEW

**Comment by mswiderski** [comment]
I think we need this commit to be backported too: https://github.com/droolsjbpm/jbpm/commit/e462da6dcb580ce604a0d2d85037f47d7c24e99e

**Comment by psiroky** [comment]
@mswiderski yeah, looks like that. Any idea why the tests are not failing currently (without the cherry-pick)?

**Comment by mswiderski** [comment]
because ejb tests are not executed as part of the build since they require EAP instance to use

**Comment by psiroky** [comment]
Ah, I see. Looks like Cargo could help in here. I think we should enable those tests by default?

I will cherry-pick the additional commit as well.

**Comment by psiroky** [comment]
@mswiderski is it safe to take the whole change? (I see there are changes than just the deploy->install).

**Comment by mswiderski** [comment]
it should be safe, though since it was not requested I don't think we should put that change in 6.3.x. Only the change in the tests as they might be used for ejb testing by QE...

**Comment by psiroky** [comment]
I've included the EJB test fixes as part of the first commit.

**Comment by mswiderski** [comment]
excellent, once the build completes I'll merge it, thanks a lot @psiroky

File: jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerTest.java [hunk 1]
```diff
@@ -18,6 +18,10 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
+import java.io.ByteArrayInputStream;
+import java.lang.reflect.Field;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
```
Comment (psiroky) [inline jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerTest.java]: These imports are result of bad merge, will remove them.

**Comment by mswiderski** [comment]
@psiroky ejb tests are arquillian tests so not sure if we can use cargo with it. But it should be enough to have EAP on jenkins and point it via system property - at least of local ejb tests https://github.com/droolsjbpm/jbpm/blob/master/jbpm-services/jbpm-services-ejb/jbpm-services-ejb-impl/src/test/resources/arquillian.xml#L8

for ejb client test cargo could be of great use as we need to deploy application to remote server and then connect to it with ejb client

**Comment by psiroky** [comment]
Ah, Arquillian :) I think we should be able to enable those in CI (even if stay disabled by default). If it is possible to enable these by just setting the property, that would be great. I've created https://issues.jboss.org/browse/JBPM-4968 to make sure I don't forget.

**Comment by mswiderski** [comment]
merged into 6.3.x

**Pull request closed without merging**